CDP4Dal/Session.cs

```diff
@@ -907,6 +907,46 @@ public async Task RefreshAuthenticationInformation()
             this.Dal.ApplyAuthenticationCredentials(this.Credentials);
         }
 
+        /// <summary>
+        /// Retrieves the  username of the authenticated user, if applicable
+        /// </summary>
+        /// <returns>An awaitable <see cref="Task{TResult}"/> with the username</returns>
+        /// <exception cref="InvalidOperationException">If credentials are not fully initialized</exception>
+        public async Task<string> QueryAuthenticatedUserName()
+        {
+            if (this.ActivePerson != null)
+            {
+                return this.ActivePerson.ShortName;
+            }
+
+            if (!this.Credentials.IsFullyInitialized)
+            {
+                throw new InvalidOperationException("Cannot retrieve authenticated User Name when credentials are fully initiliazed");

```

Comment:
Cannot retrieve authenticated User Name when credentials **not** are fully initiliazed